tutorsuperset/templates/superset/apps/pythonpath/openedx_sso_security_manager.py

```diff
@@ -19,27 +48,23 @@ def set_oauth_session(self, provider, oauth_response):
         if provider == "openedxsso":
             session["oauth_token"] = oauth_response
         return res
+    
+    def decoded_user_info(self):
+        return jwt.decode(self.access_token, algorithms=["HS256"], options={"verify_signature": False})
 
     def oauth_user_info(self, provider, response=None):
-        openedx_apis = current_app.config['OPENEDX_API_URLS']
         if provider == 'openedxsso':
-            oauth_remote = self.appbuilder.sm.oauth_remotes[provider]
-            username_url = openedx_apis['get_username']
-            me = oauth_remote.get(username_url).json()
-            username = me['username']
+            user_profile = self.decoded_user_info()
 
-            user_profile_url = openedx_apis['get_profile'].format(username=username)
-            user_profile = oauth_remote.get(user_profile_url).json()
-
-            user_roles = self._get_user_roles(username)
+            user_roles = self._get_user_roles(user_profile.get('preferred_username'))
 
             return {
                 'name': user_profile['name'],
                 'email': user_profile['email'],
-                'id': user_profile['username'],
-                'username': user_profile['username'],
-                'first_name': '',
-                'last_name': '',
+                'id': user_profile['preferred_username'],
+                'username': user_profile['preferred_username'],
+                'first_name': user_profile.get('given_name', user_profile.get('name', '')),
```

Comment:
My mistake, this doesn't work, because usually `user_profile['given_name'] == ''`.
Will have to be:
```suggestion
                'first_name': user_profile.get('given_name') || user_profile.get('name', ''),
```